tests/test_tokenization_common.py

```diff
@@ -3144,6 +3153,27 @@ def test_special_tokens_initialization(self):
                     self.assertTrue(special_token_id in p_output)
                     self.assertTrue(special_token_id in cr_output)
 
+    def test_training_new_tokenizer(self):
+        # This feature only exists for fast tokenizers
+        if not self.test_rust_tokenizer:
+            return
+
+        tokenizer = self.get_rust_tokenizer()
+        new_tokenizer = tokenizer.train_new_from_iterator(SMALL_TRAINING_CORPUS, 100)
+
+        # Test we can use the new tokenizer with something not seen during training
+        inputs = new_tokenizer(["This is the first sentence.", "This sentence is different 🤗."])
+        self.assertEqual(len(inputs["input_ids"]), 2)
```

Comment:
This test is nice but it isn't really thorough. How about testing after encoding & decoding a sequence, to ensure that the full approach works?

For example here, I'm trying the following with the ALBERT tokenizer:
```py
tokenizer.decode(inputs["input_ids"][0])
```
which returns the following sequence:
```
'[CLS]-")(")-.( as" was– the£[SEP]'
```

Now I may have misunderstood something, but I would expect a somewhat correctly tokenized/untokenized sequence. I am mistaken?

Reply:
Added a decoding test, which made me catch a problem with GPT and Herbert :-)